packages/amplify-graphql-schema-generator/src/schema-generator/generate-schema.ts

```diff
@@ -189,6 +189,15 @@ const constructObjectType = (model: Model) => {
 };
 
 const constructEnumType = (type: EnumType): EnumWrapper => {
+  // Check if the enum values imported from the database contain any invalid character.
+  // Enum can contain A-Z, a-z, 0-9 or underscore(_) and must begin with an alphabet or an underscore(_).
+  if (!validateEnumValues(type.values)) {
+    throw new Error(
+      `Enum "${type.name}" (values: ${type.values.join(
+        ',',
+      )}) contains one or more invalid values. Enum values can contain A-Z, a-z, 0-9 or underscore(_) and must begin with an alphabet or an underscore(_).`,
```

Comment:
Let's get confirmation on the error message. The prose representation of the rules is probably _less_ understandable to devs than something like "...must match the regex `[A-Za-z_][A-Za-z0-9_]+`"

Reply:
Makes sense. Updated the error message to regex.